docs/proposals/resource-aggregation-proxy/README.md

```diff
@@ -0,0 +1,455 @@
+
+![with-karmada](with-karmada.svg)
+
+We can't access pod and node information from Karmada API server. So after introducing Karmada, the PaaS components can no longer access pod and node resources. This will break the PaaS components. 
+
+
+
+In order to solve the problem, we can either:
+
+1. **Modify all of PaaS components.** Force all of them understand Karmada's logic, and find the pod node in the appropriate cluster by themselves.
+2. **Provide a Resource Aggregation Proxy.** All PaaS components just ask the proxy to give them the pod node information. The proxy will find the right clusters to get data, merge data, then return it to the client.
+
+
+
+Apparently, using the proxy will greatly simplify external components' access to kubernetes clusters, and hide multiple cluster's implementation details from them. That's what we propose.
+
+
+
+
+Try these:
+
+- **[Aggregated Kubernetes API Endpoint](https://github.com/karmada-io/karmada/blob/master/docs/userguide/aggregated-api-endpoint.md)** This component is built for Unified authentication. The restful api client still needs to enumerate all the member clusters to get the right information.
+- **[Caching](https://github.com/karmada-io/karmada/tree/master/docs/proposals/caching) (aka karmada-search)** This component can only get/list, not watch or update data. 
+
+# Motivation
+
+Write a Resource Aggregation Proxy for Karmada. This proxy provides native kubernetes API for components outside kubernetes. When the client requests operations (including reads & writes), the proxy will figure out the correct downstream kubernetes API servers (including Karmada API server and member clusters), request all the selected servers, and merge the responses into one before responding to the client. External components can treat Karmada-managed kubernetes clusters as one single cluster.
+
+## Goals
+
+Allow kubernetes API clients (kubectl, client-go, and other clients that use kubernetes APIs) to read/write the following resources:
+
+- workload: create/get/list/watch/update/patch/delete
+- node: get/list/watch/update/patch
+- pod: get/list/watch/update/patch/logs/exec
+
+## Non-Goals
+
+- read/write resources:
+  - service account token
+  - event
+
+- Provide a solution for unique resource names across all member clusters.
+
+# Proposal
+
+Add a Resource Aggregation Proxy. When the client tries to list pod by label, this proxy will aggregate pods information from all member clusters, and then return to the client as a single list. When the client tries to query workload resources, the proxy will get the information from Karmada API server directly.
+
+## User Stories
+
+### Story 1: Service Discovery
+
+Not everyone uses kubernetes Services for service discovery. We use a private service discovery architecture to accommodate both containerized and traditional applications deployed on physical machines. 
+
+![user story - service discovery](user story - service discovery.svg)
```

Comment:
It doesn't seem to be displayed properly here.

![image](https://user-images.githubusercontent.com/78244870/175901040-7b5bf9d4-62c3-42a2-a62c-52615e42d352.png)

Reply:
Updated